### PR TITLE
Stop using Vector::unsafeAppendWithoutCapacityCheck() in WasmBBQJIT.cpp

### PR DESCRIPTION
**Pull request by cdumez**
#### 41848d931422aa7edaeead93a1c57d44609fe022
<pre>
Stop using Vector::unsafeAppendWithoutCapacityCheck() in WasmBBQJIT.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=264646">https://bugs.webkit.org/show_bug.cgi?id=264646</a>

Reviewed by Darin Adler.

* Source/JavaScriptCore/wasm/WasmBBQJIT.cpp:
(JSC::Wasm::BBQJIT::addSwitch):

Canonical link: <a href="https://commits.webkit.org/270622@main">https://commits.webkit.org/270622@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/fd988128a466b06a7fdd34ee6bbb9193e3c0e7f9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25953 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4564 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27234 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28053 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23768 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6326 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1992 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/23844 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26205 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3444 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22358 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28633 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3064 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23311 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/29377 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/22605 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23678 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23688 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27254 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/25174 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3098 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1304 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/32618 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4488 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/7084 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6235 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3555 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3416 "Built successfully") | | |

### CODE REVIEW

**Comment by cdumez** [comment]
EWS is green, the patch is ready for review.

File: Source/JavaScriptCore/wasm/WasmBBQJIT.cpp [hunk 1]
```diff
@@ -7559,28 +7559,26 @@ class BBQJIT {
...
+                bool isCodeEmitted = currentControlData().addExit(*this, target->targetLocations(), results);
                 if (isCodeEmitted)
-                    targets[index]->addBranch(m_jit.jump());
+                    target->addBranch(m_jit.jump());
                 else {
                     // It is common that we do not need to emit anything before jumping to the target block.
                     // In that case, we put Box<Label> which will be filled later when the end of the block is linked.
                     // We put direct jump to that block in the link task.
-                    targets[index]->addLabel(WTFMove(label));
+                    target->addLabel(Box<CCallHelpers::Label> { label });
```
Comment (darinadler) [inline Source/JavaScriptCore/wasm/WasmBBQJIT.cpp]: I would have expected `Box { label }` here. Do we really need the template argument?

**Review by darinadler — APPROVED** [review]

File: Source/JavaScriptCore/wasm/WasmBBQJIT.cpp (hunk 1 above)
Comment (cdumez) [reply Source/JavaScriptCore/wasm/WasmBBQJIT.cpp]: I'll check, thanks.

Comment (darinadler) [reply Source/JavaScriptCore/wasm/WasmBBQJIT.cpp]: Could also experiment with leaving out the type altogether and just putting in the braces. No idea if that will work.

**Pull request merged**